### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [wwmm/pulseeffects#617]
Would you like to add more error handling for return values from functions like the following?
- [bindtextdomain](https://linux.die.net/man/3/bindtextdomain "Set directory containing message catalogs.") ⇒ [main](https://github.com/wwmm/pulseeffects/blob/b0c52fa2f52c8a22b3b61be3cc47950afaa6efaf/src/pulseeffects.cpp#L18)
- [strdup](https://pubs.opengroup.org/onlinepubs/9699919799/functions/strdup.html "Duplicate a string.") ⇒ [gst_pecrystalizer_class_init](https://github.com/wwmm/pulseeffects/blob/8e853108b3bdb0e61bc1ea13a0c6dc907ef4fcce/src/crystalizer/gstpecrystalizer.cpp#L151)

**Comment by wwmm** [comment]
No problems.

**Comment by wwmm** [comment]
done

**Comment by elfring** [comment]
I suggest to reconsider further implementation details.
How do you think about to improve [static source code analysis](https://en.wikipedia.org/wiki/List_of_tools_for_static_code_analysis#C.2C_C.2B.2B "Tools for code analysis") also for this software?

**Comment by wwmm** [comment]
I use clang static analysis through a vscode plugin. It seems enough.

**Comment by elfring** [comment]
I find the error detection and corresponding exception handling still incomplete.

**Comment by elfring** [comment]
Will the [desire](https://cwe.mitre.org/data/definitions/252.html "Checking more return values") grow to tackle the remaining update candidates?

**Comment by wwmm** [comment]
Yes. But with my vacations ending time will be shorter. In any case feel free to open issues with what you find. Whenever possible I will try to improve the situation :-)

**Comment by elfring** [comment]
Would you like to reopen this issue until it will be fixed completely?

**Comment by wwmm** [comment]
Ok. In the case of the catch all in the main function how is the message supposed to be printed?

**Comment by elfring** [comment]
The clarification around the application of catch clauses will matter more for an other issue.

Do mentioned source code analysis tools point still open issues out because of return value ignorance and missing memory releases?

**Comment by wwmm** [comment]
Good question. I use this https://code.visualstudio.com/docs/languages/cpp and up to now I did not try to change static code analysis settings. I only created a file for clang-format. I will take a look and see if there is a way to be informed about ignored function return values.

**Comment by wwmm** [comment]
Based on what I saw here https://github.com/Microsoft/vscode-cpptools/issues/783 clang is used only for the format. The static analysis is made by a built-in code. I will see how to configure it.

**Comment by wwmm** [comment]
Up to now I saw only a setting to make vscode intellisense to use clang. But it is not clear if this affects the static analysis. And I still did not find any way to configure what is or not analyzed =/

**Comment by wwmm** [comment]
I found another extension that is based on clang tidy https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd. I will spend some time configuring it.

**Comment by wwmm** [comment]
Now I have a working clang-tidy setup :-). Configuration file here https://github.com/wwmm/pulseeffects/blob/master/.clang-tidy. The next step is to decide which warnings make sense. The one in the following image won't be productive
![Screenshot from 2020-01-31 21-36-59](https://user-images.githubusercontent.com/16225172/73583791-d775bf00-4472-11ea-9c68-a4bc03db657b.png)
GStreamer is a C library without a feature full C++ wrapper. The easier way to deal with it is using its typecasting functions. I will see how to disable this warning.

**Comment by elfring** [comment]
How did [the GStreamer API C++ bindings](https://gstreamer.freedesktop.org/bindings/cplusplus.html "Bindings for C++") evolve so far?

**Comment by wwmm** [comment]
Current GStreamer version is `1.16.2` and the c++ wrapper is still in `1.10`. Not good. It doesn't have all C functions exposed. There are entire libraries like `gst-plugins-bad` that are not there. When I began to write PulseEffects I used Python. Its GStreamer wrapper is pretty good. But even there there were GStreamer functions not exposed by the wrapper. When I ported to c++ I thought a lot about the best course of action. Talking to C directly was the path with less headaches.
